components/board/pi/common/common.go

```diff
@@ -17,23 +17,28 @@ var ModelName = resource.NewDefaultModel("pi")
 
 // ServoConfig is the config for a pi servo.
 type ServoConfig struct {
-	Pin      string   `json:"pin"`
-	Min      int      `json:"min,omitempty"`
-	Max      int      `json:"max,omitempty"`
-	StartPos *float64 `json:"starting_position_degs,omitempty"`
-	HoldPos  *bool    `json:"hold_position,omitempty"` // defaults True. False holds for 500 ms then disables servo
+	Pin       string   `json:"pin"`
+	Min       int      `json:"min,omitempty"`
+	Max       int      `json:"max,omitempty"`
+	StartPos  *float64 `json:"starting_position_degs,omitempty"`
+	HoldPos   *bool    `json:"hold_position,omitempty"` // defaults True. False holds for 500 ms then disables servo
+	BoardName string   `json:"board"`
 }
 
 // Validate ensures all parts of the config are valid.
-func (config *ServoConfig) Validate(path string) error {
+func (config *ServoConfig) Validate(path string) ([]string, error) {
+	var deps []string
 	if config.Pin == "" {
-		return utils.NewConfigValidationError(path,
+		return deps, utils.NewConfigValidationError(path,
```

Comment:
Discussion question: this seems correct for now, but returning a variable here feels like it might not be future-proof: maybe someone will initialize `deps` to something that's not `nil`, and then this will give confusing results (we would return a non-empty array and a non-nil error). Would it make more sense to return `nil` instead of `deps` for these error cases?

Reply:
:shrug: Maybe not, this PR LGTM as-is, too.

Reply:
Thanks! that's a good observation. Passing nil instead of empty array makes sense.